auth.php

```diff
@@ -164,18 +164,35 @@ public function user_login ($username, $password) {
     }
 
     /**
-     * Check if the password is master password.
      * @param $userpassword
      * @return bool
+     * @throws dml_exception
      */
     private function is_master_password($userpassword) {
-        global $CFG;
-        if (isset($CFG->forced_plugin_settings)) {
-            if ($CFG->forced_plugin_settings["auth_basic"]) {
-                if ($masterpassword = $CFG->forced_plugin_settings["auth_basic"]['master']) {
-                    return $masterpassword === $userpassword;
+        global $CFG, $DB;
+        if (isset($CFG->auth_basic_enabled_master_password) && $CFG->auth_basic_enabled_master_password == true) {
+            $remoteaddress = getremoteaddr('0.0.0.0');
```

Comment:
the 0.0.0.0 is the default so redundant. Also it's not even needed for the remoteip_in_list call so to make it more obvious I'd move this down to just before it is used in the ->log call